src/rgw/rgw_rest_s3.cc

```diff
@@ -3086,12 +3214,21 @@ RGWOp *RGWHandler_REST_Service_S3::op_post()
   return NULL;
 }
 
-RGWOp *RGWHandler_REST_Bucket_S3::get_obj_op(bool get_data)
+RGWOp *RGWHandler_REST_Bucket_S3::get_obj_op(bool get_data, int list_type)
 {
   // Non-website mode
   if (get_data) {
-    return new RGWListBucket_ObjStore_S3;
-  } else {
+    if(list_type == 1)
```

Comment:
`int list_type` needs to come from the http params that are in `s->info.args`. see `RGWHTTPArgs::get_int()` to read the "list-type" param as an int

Reply:
Could you take a look at it?

Reply:
yeah, the default should be v1, and we should do v1 for any value other than 2 here